extensions/wikia/UserProfilePageV3/UserIdentityBoxDiscussionInfo.class.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+use Swagger\Client\ApiException;
+use Swagger\Client\Discussion\Api\ContributionApi;
+use Swagger\Client\Discussion\Api\SitesApi;
+use Wikia\DependencyInjection\Injector;
+use Wikia\Service\Swagger\ApiProvider;
+
+class UserIdentityBoxDiscussion {
+
+	const DISCUSSION_SERVICE_NAME = 'discussion';
+	const TIMEOUT = 5;
+
+	private $userId = null;
+	private $discussionActive = false;
+	private $discussionPostsCount = 0;
+	private $discussionAllPostsByUserLink = '';
+
+	function __construct($user) {
+		$this->userId = $user->getId();
+	}
+
+	public function isDiscussionActive() {
+		return $this->discussionActive;
+	}
+
+	public function getDiscussionPostsCount() {
+		return $this->discussionPostsCount;
+	}
+
+	public function getDiscussionAllPostsByUserLink() {
+		return $this->discussionAllPostsByUserLink;
+	}
+
+	public function fetchDiscussionPostsNumber() {
+		$siteId = F::app()->wg->CityId;
+
+		$this->discussionActive = $this->checkDiscussionActive($siteId);
+		if ( $this->discussionActive ) {
+			$this->discussionPostsCount = $this->fetchDiscussionPostsCount($siteId);
+			$this->discussionAllPostsByUserLink = "/d/u/{$this->userId}";
+		}
+	}
+
+	private function checkDiscussionActive($siteId) {
+		try {
+			$this->getDiscussionSitesApi()->getSite( $siteId );
+			return true;
+		} catch ( ApiException $e ) {
```

Comment:
Code was based on `SpecialDiscussionsController.class.php`, which handles errors these way. I could only confirm that 404 responses are caught.